### PR TITLE
Add native Processing widget wrapper for raster creation options

### PR DESCRIPTION
**Pull request by alexbruy**
## Description

Add native Processing widget wrapper for raster creation options which replaces custom widget wrapper living in the GDAL Processing provider codebase. The new wrapper with the "rasteroptions" `parameterType` can be used together with string parameters to provide user with a widget for editing GDAL raster creation options. This allows to use raster options widget not only with GDAL algorithms but with any Processing algorithm where this makes sense.

Also update GDAL algorithms to use native widget wrapper and remove old implementation.

### CODE REVIEW

File: src/gui/processing/qgsprocessingrasteroptionswidgetwrapper.cpp [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+void QgsProcessingRasterOptionsWidgetWrapper::setWidgetValue( const QVariant &value, QgsProcessingContext &context )
+{
+  QString val = QgsProcessingParameters::parameterAsString( parameterDefinition(), value, context );
+
+  if ( mOptionsWidget )
+  {
+    if ( value.isValid() )
+      mOptionsWidget->setOptions( val );
```
Comment (nyalldawson) [inline src/gui/processing/qgsprocessingrasteroptionswidgetwrapper.cpp:80]: The original python code had logic here to change pipe separated options to space separated -- is that still required?

File: tests/src/gui/testprocessinggui.cpp [hunk 2]
```diff
@@ -10054,6 +10057,30 @@ void TestProcessingGui::testAlignRasterLayersWrapper()
...
+
+  std::unique_ptr<QWidget> w( wrapper.createWidget() );
+  QVERIFY( w );
+
+  QSignalSpy spy( &wrapper, &QgsProcessingRasterOptionsWidgetWrapper::widgetValueHasChanged );
+  wrapper.setWidgetValue( QStringLiteral( "TFW=YES" ), context );
+  QCOMPARE( spy.count(), 1 );
+  QCOMPARE( wrapper.widgetValue().toString(), QStringLiteral( "TFW=YES" ) );
+  wrapper.setWidgetValue( QStringLiteral( "TFW=YES TILED=YES" ), context );
+  QCOMPARE( wrapper.widgetValue().toString(), QStringLiteral( "TFW=YES|TILED=YES" ) );
```
Comment (nyalldawson) [inline tests/src/gui/testprocessinggui.cpp:10081]: I think we need a test here with setWidgetValue with pipe separated options too

Comment (alexbruy) [reply tests/src/gui/testprocessinggui.cpp:10081]: Added.

**Review by nyalldawson — APPROVED** [review]

**Comment by agiudiceandrea** [comment]
Is this somehow an API break? i.e. I think that plugins using
```python
        options_param.setMetadata({
            'widget_wrapper': {
                'class': 'processing.algs.gdal.ui.RasterOptionsWidget.RasterOptionsWidgetWrapper'}})
        self.addParameter(options_param)
```
will not correctly work using QGIS >= 3.40.

**Pull request merged**